cuqi/mcmc/_mh.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+import cuqi
+from cuqi.mcmc import ProposalBasedSamplerNew
+from cuqi.array import CUQIarray
+
+
+class MHNew(ProposalBasedSamplerNew):
+    """ Metropolis-Hastings (MH) sampler.
+
+    Parameters
+    ----------
+    target : cuqi.density.Density
+        Target density or distribution.
+
+    proposal : cuqi.distribution.Distribution or callable
+        Proposal distribution. If None, a random walk MH is used (i.e., Gaussian proposal with identity covariance).
+
+    scale : float
+        Scaling parameter for the proposal distribution.
+
+    kwargs : dict
+        Additional keyword arguments to be passed to the base class :class:`ProposalBasedSamplerNew`.
+
+    """
+
+    def __init__(self, target, proposal=None, scale=1, **kwargs):
+        super().__init__(target, proposal=proposal, scale=scale, **kwargs)
+
+    def validate_target(self):
+        pass # All targets are valid
+
+    @ProposalBasedSamplerNew.proposal.setter  # TODO. Check if we can refactor this. We can work with a validate_proposal method instead?
```

Comment:
Not sure I understand this? Why is the setter referring to its superclass proposal method?

Reply:
This is "legacy" copy-paste from the old sampler implementation. It overwrites the setter for proposal

Reply:
Its added as issue in #378